editor/components/inserter/menu.js

```diff
@@ -12,11 +13,59 @@ import Dashicon from 'components/dashicon';
 class InserterMenu extends wp.element.Component {
 	constructor() {
 		super( ...arguments );
+		this.blockTypes = wp.blocks.getBlocks();
+		this.categories = wp.blocks.getCategories();
```

Comment:
This assumes the categories and blocks don't change over time. It's true for now, we can keep it this way. But worth nothing this could change.